core/src/main/java/bisq/core/app/AppStartupState.java

```diff
@@ -0,0 +1,149 @@
+            @Override
+            public void onUpdatedDataReceived() {
+                updatedDataReceived.set(true);
+            }
+        });
+
+        walletsSetup.downloadPercentageProperty().addListener((observable, oldValue, newValue) -> {
+            if (walletsSetup.isDownloadComplete())
+                isBlockDownloadComplete.set(true);
+        });
+
+        walletsSetup.numPeersProperty().addListener((observable, oldValue, newValue) -> {
+            if (walletsSetup.hasSufficientPeersForBroadcast())
+                hasSufficientPeersForBroadcast.set(true);
+        });
+
+        p2pNetworkAndWalletInitialized = EasyBind.combine(updatedDataReceived,
+                isBlockDownloadComplete,
+                hasSufficientPeersForBroadcast,
+                allDomainServicesInitialized,
+                (a, b, c, d) -> {
+                    if (a && b && c) {
+                        walletAndNetworkReady.set(true);
+                    }
+                    return a && b && c && d;
+                });
+        p2pNetworkAndWalletInitialized.subscribe((observable, oldValue, newValue) -> {
+            if (newValue) {
+                applicationFullyInitialized.set(true);
+            }
+        });
+    }
+
+    public void onDomainServicesInitialized() {
+        allDomainServicesInitialized.set(true);
+    }
+
+
+    ///////////////////////////////////////////////////////////////////////////////////////////
+    // Getters
+    ///////////////////////////////////////////////////////////////////////////////////////////
+
+    public boolean isWalletAndNetworkReady() {
+        return walletAndNetworkReady.get();
+    }
+
+    public ReadOnlyBooleanProperty walletAndNetworkReadyProperty() {
+        return walletAndNetworkReady;
+    }
+
+    public boolean isAllDomainServicesInitialized() {
+        return allDomainServicesInitialized.get();
+    }
+
+    public ReadOnlyBooleanProperty allDomainServicesInitializedProperty() {
+        return allDomainServicesInitialized;
+    }
+
+    public boolean isApplicationFullyInitialized() {
+        return applicationFullyInitialized.get();
+    }
+
+    public ReadOnlyBooleanProperty applicationFullyInitializedProperty() {
+        return applicationFullyInitialized;
+    }
+
+    public boolean isUpdatedDataReceived() {
+        return updatedDataReceived.get();
+    }
+
+    public ReadOnlyBooleanProperty updatedDataReceivedProperty() {
+        return updatedDataReceived;
+    }
+
+    public boolean isIsBlockDownloadComplete() {
+        return isBlockDownloadComplete.get();
+    }
+
+    public ReadOnlyBooleanProperty isBlockDownloadCompleteProperty() {
+        return isBlockDownloadComplete;
+    }
+
+    public boolean isHasSufficientPeersForBroadcast() {
+        return hasSufficientPeersForBroadcast.get();
+    }
+
+    public ReadOnlyBooleanProperty hasSufficientPeersForBroadcastProperty() {
+        return hasSufficientPeersForBroadcast;
+    }
```

Comment:
These look unused. Am I missing something,

Reply:
Yes, they are there for future usage. Some might not be needed, once we have refactored it completely we might remove some, but better to have it in place now so we can start migrating. 
At the checks for when to start a domain specific init is duplicated in many places and often not correct. This PR should start to refactor those cases and avoid those issues.